examples/settings/settings_main.c

```diff
@@ -0,0 +1,276 @@
+
+  ret = settings_init();
+  if (ret < 0)
+    {
+      printf("settings init failed: %d", ret);
+      goto end;
+    }
+
+  ret = settings_setstorage(path, storage_type);
+  if (ret == -ENOENT)
+    {
+      printf("INFO: no existing binary settings file found. Creating it.\n");
+      fd = open(path, O_CREAT);
+      if (fd < 0)
+        {
+          printf("failed to create settings file\n");
+          goto end;
+        }
+
+      close(fd);
+    }
+  else if (ret < 0)
+    {
+      printf("settings setstorage failed: %d\n", ret);
+      goto end;
+    }
+  else
+    {
+      printf("INFO: a settings file was found and loaded\n");
+      ret = settings_get("v1", SETTING_STRING, readstr,
+                         CONFIG_SETTINGS_VALUE_SIZE);
+      if (ret < 0)
+        {
+          printf("settings retrieve failed: %d\n", ret);
+          goto end;
+        }
+
+      printf("Retrieved settings value (v1) with value:%s\n",
+              readstr);
+    }
+
+  printf("Creating an integer settings value (v1) with value:0x%x\n",
+          testval);
+  ret = settings_create("v1", SETTING_INT, testval);
+  if (ret < 0)
+    {
+      printf("settings create failed: %d\n", ret);
+      goto end;
+    }
+
+  ret = settings_type("v1", &stype);
+  if (ret < 0)
+    {
+      printf("failed to read settings type: %d\n", ret);
+      goto end;
+    }
+
+  printf("Retrieved setting type is:%d\n", stype);
+
+  ret = settings_get("v1", SETTING_INT, &testval);
+  if (ret < 0)
+    {
+      printf("settings retrieve failed: %d\n", ret);
+      goto end;
+    }
+
+  printf("Retrieved integer settings value (v1) with value:0x%x\n",
+          testval);
+
+  ret = settings_get("v1", SETTING_BOOL, &testval);
+  if (ret < 0)
+    {
+      printf("Deliberate fail: BOOL type requested not INT. Error:%d\n",
+              ret);
+    }
+
+  ret = settings_get("v2", SETTING_INT, &testval);
+  if (ret < 0)
+    {
+      printf("Deliberate fail: non-existent setting requested. Error:%d\n",
+             ret);
+    }
+
+  teststr = "I'm a string!";
+  printf("Changing an integer settings value (v1) to string:%s\n",
+          teststr);
+  ret = settings_create("v1", SETTING_STRING, teststr);
+  if (ret < 0)
+    {
+      printf("settings changed failed: %d\n", ret);
+      goto end;
+    }
+
+  ret = settings_get("v1", SETTING_STRING, &readstr,
+                     CONFIG_SETTINGS_VALUE_SIZE);
+  if (ret < 0)
+    {
+      printf("settings retrieve failed: %d\n", ret);
+      goto end;
+    }
+
+  printf("Retrieved string settings value (v1) with value:%s\n",
+          readstr);
+
+  FAR struct in_addr save_ip;
+  FAR struct in_addr load_ip;
+  save_ip.s_addr = HTONL(0xc0a86401);
+  ret = settings_create("IP0", SETTING_IP_ADDR, &save_ip);
+  if (ret < 0)
+    {
+      printf("IP address settings create failed: %d\n", ret);
+      goto end;
+    }
+
+  printf("Creating an IP settings value (IP0) with value:0x%08lx\n",
+          save_ip.s_addr);
```

Comment:
```suggestion

  printf("Creating an IP settings value (IP0) with value:0x%08lx\n",
          save_ip.s_addr);

  FAR struct in_addr save_ip;
  FAR struct in_addr load_ip;
  save_ip.s_addr = HTONL(0xc0a86401);
  ret = settings_create("IP0", SETTING_IP_ADDR, &save_ip);
  if (ret < 0)
    {
      printf("IP address settings create failed: %d\n", ret);
      goto end;
    }

```

Maybe  move the print before actually attempting the call?

Reply:
Yep, got swapped around when I tweaked it I think.